src/Decompiler/Scanning/BlockWorkitem.cs

```diff
@@ -1083,11 +1083,8 @@ private Block FallthroughBlock(Address addrSrc, Procedure proc, Address fallthru
         private Block AddIntraStatementBlock(Procedure proc)
         {
             var label = ric.Address.GenerateName("l", string.Format("_{0}", ++extraLabels));
-            var fallthru = new Block(proc, label)
-            {
-                IsSynthesized = true
-            };
-            proc.ControlGraph.Blocks.Add(fallthru);
+            var fallthru = proc.AddBlock(ric.Address, label);
+            fallthru.IsSynthesized = true;
```

Comment:
Let's add an `AddSyntheticBlock` to `Procedure`.

Reply:
Done in 3e9e27c